6.5.group_by.py

```diff
@@ -0,0 +1,14 @@
+
+
+def group_by(func, iterable):
+    """
+    The code defines a function called group_by that takes two parameters:
+    a function func and an iterable iterable.
+    The function returns a dictionary where the keys are the values returned
+    by the func function when applied to each element of iterable,
+    and the values are lists of the elements that share the same key.
+    """
+    return {func(item): [i for i in iterable if func(i) == func(item)] for item in iterable}
```

Comment:
Cool dict-list comprehension!
Notice that in real-life code we will prefer to not write a one-liner like this, because it is very hard to actually understand how it works.
Here it's ok because it's a comprehension exercise :)